### PR TITLE
Support for accessors (#99)

### PR DESCRIPTION
**Pull request by lukasbach**
The changes introduce the documentation of accessors (both get and set) for typescript classes, as described in the issue #99. If you have further suggestions or change requests for this PR, please tell me and I will adapt the code.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/documentalist](https://github.com/palantir/documentalist), @lukasbach! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/compiler/src/__tests__/__snapshots__/typescript.test.ts.snap [hunk 1]
```diff
@@ -201,18 +232,49 @@ limitations under the License.",
...
     "properties": Array [],
     "sourceUrl": "https://github.com/palantir/documentalist/blob/develop/packages/compiler/src/__tests__/__fixtures__/classes.ts#L18",
   },
   "Dog": Object {
+    "accessors": Array [
+      Object {
+        "documentation": Object {
+          "contents": Array [
+            "<p>Get the noise of the animal
+Set the noise for the animal</p>
```
Comment (adidahiya) [inline packages/compiler/src/__tests__/__snapshots__/typescript.test.ts.snap]: Is it possible to separate these two JSDoc blocks into different paragraphs?

File: packages/client/src/typescript.ts [hunk 2]
```diff
@@ -109,6 +110,14 @@ export interface ITsConstructor extends ITsDocBase, ITsCallable {
     kind: Kind.Constructor;
 }
 
+export interface ITsAccessor extends ITsDocBase {
+    kind: Kind.Accessor;
+    /** Value type for the set accessor, or null if set accessor is not defined. */
+    valueType: string | null;
```
Comment (adidahiya) [inline packages/client/src/typescript.ts]: get and set accessors are required to have the same type, so just one `type: string` is necessary here. and then we don't need to handle the undefined case.

**Comment by lukasbach** [comment]
@adidahiya Thank you for your feedback. I will be unavailable for the coming week, and work on the requested changes after that.

**Comment by lukasbach** [comment]
@adidahiya I updated the PR with the requested changes. The documentation object still contains the merged docs for the get- and set-accessor (as a "documentation" object for an accessor with potentially two signatures is not clearly defined), but two additional properties ``getDocumentation`` and ``setDocumentation`` describe the docs for the respective signatures. Also, as you've suggested, an accessor now only contains a single type.

File: packages/client/src/typescript.ts [hunk 3]
```diff
@@ -109,6 +110,16 @@ export interface ITsConstructor extends ITsDocBase, ITsCallable {
...
 }
 
+export interface ITsAccessor extends ITsDocBase {
+    kind: Kind.Accessor;
+    /** If a set signature is defined and documented for this accessor, this will contain its documentation. */
+    getDocumentation: IBlock | undefined;
+    /** If a get signature is defined and documented for this accessor, this will contain its documentation. */
+    setDocumentation: IBlock | undefined;
+    /** Type of the accessor, or null if neither a get or a set accessor is defined. */
+    type: string | null;
```
Comment (adidahiya) [inline packages/client/src/typescript.ts]: when would this actually be `null`? why would we ever generate a `ITsAccessor` object in that case?

Comment (lukasbach) [reply packages/client/src/typescript.ts]: You're right, sorry that was from my code before where I had seperate declarations for get and set accessors. Changed in the updated PR.

**Comment by adidahiya** [comment]
thanks for the PR @lukasbach!

**Pull request merged**